tests/rptest/tests/datalake/datalake_services.py

```diff
@@ -53,7 +60,7 @@ def setUp(self):
 
         self.catalog_service.start()
 
-        if not self.catalog_service.filesystem_wrapper_mode:
+        if not self.catalog_service.catalog_type() == CatalogType.REST_HADOOP:
```

Comment:
So IIUC, if the catalog type is the HadoopCatalog, the warehouse name should be equal to iceberg_catalog_base_location, and data will be placed under <bucket>/<iceberg_catalog_base_location>. Spark is expected to use spark_catalog_name.warehouse=s3a://<bucket>/<iceberg_catalog_base_location>

If the catalog type is a REST catalog, the warehouse name should be equal to iceberg_rest_catalog_prefix. Spark is expected to use spark_catalog_name.warehouse=<iceberg_rest_catalog_prefix>.

Does that align with your understanding? I'm thinking that for ducktape, we should pass the warehouse_name to both iceberg_catalog_base_location and iceberg_rest_catalog_prefix cluster configs.

Then, all CatalogServices will have a roughly uniform concept of what a warehouse is.